### ISSUE TITLE
Create cross-platform build script

### ISSUE THREAD

**Issue by janusw** [Mapsui/Mapsui#1135]
The current build script (`Scripts/buildpack.bat`) only works on Windows, but not on MacOS.

In order to be able to build from the command line and create nuget packages also on MacOS, one could either:
* add a Mac-specific build script or
* create a cross-platform build script (maybe based on [cake](https://cakebuild.net/))

Option 2 is clearly preferable IMHO, but option 1 is still better than the current state of things.

**Comment by janusw** [comment]
One limitation of the build procedure on Mac is that it cannot build the Windows-specific parts (WPF and UWP), but everything else should actually be transferable.

I am starting to work on a Mac-specific build script in a first step. Once this is done, one could possibly meld the platform-specific scripts into a cake-based cross-platform script.

**Comment by pauldendulk** [comment]
Yes, this needs improvements. Not sure which is the best way to proceed. I prefer a single script of course, but the main problem is not the script but the fact that we have components that can only be build on 1 of the platforms.

Not inclined to add Cake. Linux sh scripts are actually not that bad because they could also be built with WSL2 on Windows. I expect this to become a more common approach in general.

Eventually I would like to use more github actions and also release nugets through github actions.

**Comment by pauldendulk** [comment]
I also consider dropping support for things that can not be build cross platform, or move them to a separate github repository.

**Comment by janusw** [comment]
> I also consider dropping support for things that can not be build cross platform, or move them to a separate github repository.

Huh, that sounds problematic: UWP cannot be built on Mac, iOS cannot be build on Windows (without a Mac build server). I would not want to deprecate or remove any of those two.

Instead I think that for each platfrom one should only build the supported components.

For auto-building the nupkg one should stick to Windows (plus Mac build server), since that config basically supports all components.

**Comment by pauldendulk** [comment]
Valid points, still I am thinking of a way to make it more maintainable. With your new PR we have two nuspecs. The nuspec is typically something we (meaning mostly me 🙄) forget to update. Now we have two of those. What if there is one nuspec for the shared functionality and separate nuspecs for Mac and UWP/WPF?

Also it is also possible to work without nuspecs, use the solution to decide what should be in the nuget.

**Comment by janusw** [comment]
> Also it is also possible to work without nuspecs, use the solution to decide what should be in the nuget.

I was also thinking in that direction, but it might require some restructuring of the projects in the VS solution.

**Comment by janusw** [comment]
> Not inclined to add Cake. Linux sh scripts are actually not that bad because they could also be built with WSL2 on Windows. I expect this to become a more common approach in general.

What's your argument against cake?

The problem with shell scripts is that they are not 100% portable. First of all the shell on Mac is not exactly the same as on Linux/WSL (zsh vs bash), and then there are issues like file paths etc. I don't have too much experience with cake, but I think it helps to abstract away such differences, and really have a single cross-platform build script.


> Eventually I would like to use more github actions and also release nugets through github actions.

I agree that it would make sense to use github actions for the CI builds. This also came to my mind, and it might help to work around #1115.
